core/src/main/scala/sttp/openai/OpenAi.scala

```diff
@@ -20,6 +22,27 @@ class OpenAi(authToken: String) {
       .get(OpenAIEndpoints.FilesEndpoint)
       .response(asJsonSnake[FilesResponse])
 
+  /** Upload a file that contains document(s) to be used across various endpoints/features. Currently, the size of all the files uploaded by
+    * one organization can be up to 1 GB. Please contact OpenAI if you need to increase the storage limit.
+    * @param file
+    *   [[java.io.File File]] of the JSON Lines file to be uploaded. <p> If the purpose is set to "fine-tune", each line is a JSON record
+    *   with "prompt" and "completion" fields representing your
+    *   [[https://platform.openai.com/docs/guides/fine-tuning/prepare-training-data training examples]].
+    * @param purpose
+    *   The intended purpose of the uploaded documents. <p> Use "fine-tune" for Fine-tuning. This allows OpenAI to validate the format of
+    *   the uploaded file.
+    * @return
+    *   Uploaded file's basic information.
+    */
+  def uploadFile(file: File, purpose: String = "fine-tune"): Request[Either[ResponseException[String, Exception], FileData]] =
```

Comment:
`suggestion` How about we overload this method with another one that accept String as an argument, that is a path to a file?
@kciesielski @adamw

Reply:
I think it might be useful, similarly to Tapir where we have endpoints that take a `systemPath: String`.

Reply:
Okay, so here's the deal. [We cannot overload a method with default parameters](https://stackoverflow.com/questions/4652095/why-does-the-scala-compiler-disallow-overloaded-methods-with-default-arguments). So from my point of view we could either get rid of `def uploadFile(file: File, purpose: String = "fine-tune")` and leave `def uploadFile(systemPath: String, purpose: String = "fine-tune")` alone or create four methods e.g.
```
def uploadFile(file: File, purpose: String)
def uploadFile(file: File) // with hard-coded 'purpose' as 'fine-tune'
def uploadFile(systemPath: String, purpose: String)
def uploadFile(systemPath: String) // with hard-coded 'purpose' as 'fine-tune'
```
that will handle all possible options. Wdyt?

Reply:
yes that's the way to do it :)

Reply:
@kciesielski @adamw could you please take a look?

Reply:
@benzwreck +1 to Adam, use multiple method variants instead of default parameters.